internal/app/machined/pkg/controllers/k8s/control_plane_static_pod.go

```diff
@@ -203,53 +205,82 @@ func (ctrl *ControlPlaneStaticPodController) manageAPIServer(ctx context.Context
+		"proxy-client-key-file":              filepath.Join(constants.KubernetesAPIServerSecretsDir, "front-proxy-client.key"),
+		"enable-bootstrap-token-auth":        "true",
+		"tls-cipher-suites":                  "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256", //nolint:lll
+		"encryption-provider-config":         filepath.Join(constants.KubernetesAPIServerSecretsDir, "encryptionconfig.yaml"),
+		"audit-policy-file":                  filepath.Join(constants.KubernetesAPIServerSecretsDir, "auditpolicy.yaml"),
+		"audit-log-path":                     "-",
+		"audit-log-maxage":                   "30",
+		"audit-log-maxbackup":                "3",
+		"audit-log-maxsize":                  "50",
+		"profiling":                          "false",
+		"etcd-cafile":                        filepath.Join(constants.KubernetesAPIServerSecretsDir, "etcd-client-ca.crt"),
+		"etcd-certfile":                      filepath.Join(constants.KubernetesAPIServerSecretsDir, "etcd-client.crt"),
+		"etcd-keyfile":                       filepath.Join(constants.KubernetesAPIServerSecretsDir, "etcd-client.key"),
+		"etcd-servers":                       strings.Join(cfg.EtcdServers, ","),
+		"kubelet-client-certificate":         filepath.Join(constants.KubernetesAPIServerSecretsDir, "apiserver-kubelet-client.crt"),
+		"kubelet-client-key":                 filepath.Join(constants.KubernetesAPIServerSecretsDir, "apiserver-kubelet-client.key"),
+		"secure-port":                        strconv.FormatInt(int64(cfg.LocalPort), 10),
+		"service-account-issuer":             cfg.ControlPlaneEndpoint,
+		"service-account-key-file":           filepath.Join(constants.KubernetesAPIServerSecretsDir, "service-account.pub"),
+		"service-account-signing-key-file":   filepath.Join(constants.KubernetesAPIServerSecretsDir, "service-account.key"),
+		"service-cluster-ip-range":           strings.Join(cfg.ServiceCIDRs, ","),
+		"tls-cert-file":                      filepath.Join(constants.KubernetesAPIServerSecretsDir, "apiserver.crt"),
+		"tls-private-key-file":               filepath.Join(constants.KubernetesAPIServerSecretsDir, "apiserver.key"),
+		"kubelet-preferred-address-types":    "InternalIP,ExternalIP,Hostname",
 	}
 
 	if cfg.CloudProvider != "" {
-		args = append(args, fmt.Sprintf("--cloud-provider=%s", cfg.CloudProvider))
+		builder.Set("cloud-provider", cfg.CloudProvider)
 	}
 
-	for k, v := range cfg.ExtraArgs {
-		args = append(args, fmt.Sprintf("--%s=%s", k, v))
+	mergePolicies := argsbuilder.MergePolicies{
+		"enable-admission-plugins": argsbuilder.MergeAdditive,
+		"authorization-mode":       argsbuilder.MergeAdditive,
+		"tls-cipher-suites":        argsbuilder.MergeAdditive,
+		"etcd-servers":             argsbuilder.MergeAdditive,
```

Comment:
probably `MergeDenied`, as if we lock down the etcd certs, the only etcd server is the one Talos builds?